troncos/traces/decorate.py

```diff
@@ -213,6 +229,8 @@ def my_function():
 
     # End of module
     """
+    if resource:
+        attributes["resource"] = resource
```

Comment:
Attributes can be None. Just pass `resource` into `_trace_function` below and be done with it.